backoffice/views.py

```diff
@@ -256,40 +257,69 @@ def user_list(request):
         })
 
 
+def ban_user(request, user):
+    user_account, _created = UserStanding.objects.get_or_create(
+        user=user, defaults={'changed_by': request.user})
+    if request.POST['value'] == 'disable':
+        user_account.account_status = UserStanding.ACCOUNT_DISABLED
+        messages.success(request, _(u"Successfully disabled {}'s account").format(user.username))
+    elif request.POST['value'] == 'reenable':
+        user_account.account_status = UserStanding.ACCOUNT_ENABLED
+        messages.success(request, _(u"Successfully reenabled {}'s account").format(user.username))
+    user_account.changed_by = request.user
+    user_account.standing_last_changed_at = timezone.now()
+    user_account.save()
+
+def change_password(request, user):
+    user.set_password(request.POST['new-password'])
+    user.save()
+    messages.success(request, _(u"User password changed"))
+
+def change_grade(request, user):
+    """ Change a certificate grade per user, per course.
+
+    Grade comes as string and needs to range from 0 to 1.
+    Args:
+         request (HttpRequest): Contains the course id in request.POST['course-id'].
+         user (User): The user attached to the certificate.
+     """
+    course = get_course_key(request.POST['course-id'])
+    generated_certificate = GeneratedCertificate.objects.get(user=user, course_id=course)
+    try:
+        if not 0 <= float(request.POST['new-grade']) <= 1:
+            messages.error(request, _(u"Grades range from 0 to 1."))
+        else:
+            generated_certificate.grade = request.POST['new-grade']
+            generated_certificate.save()
+            messages.success(request, _(u"User grade changed."))
+    except ValueError:
```

Comment:
Le seul endroit qui peut lever une ValueError est la conversion `float(request.POST['new-grade'])`, n'est-ce pas ? Si oui, il vaut probablement mieux limiter la portion de code couverte par `try`.

    try:
        new_grade = float(request.POST['new-grade'])
    except ValueError:
        message.error(...)
    else:
        if not 0 <= new_grade <= 1:
            ...
        else:
            ...